packages/app/src/systems/SignUp/pages/WelcomeScreen/WelcomeScreen.tsx

```diff
@@ -1,12 +1,21 @@
 import { Button, Flex, FuelLogo, Stack } from '@fuel-ui/react';
+import { useEffect } from 'react';
 import { useNavigate } from 'react-router-dom';
 
 import { Header } from '../../components';
 
 import { Layout, Pages } from '~/systems/Core';
+import { useIsSigningUp } from '~/systems/Core/hooks/useIsSigningUp';
 
 export function WelcomeScreen() {
   const navigate = useNavigate();
+  const isSigningUp = useIsSigningUp();
+
+  useEffect(() => {
+    if (isSigningUp) {
+      navigate(Pages.signUpCreateWallet());
+    }
+  }, [isSigningUp]);
```

Comment:
Why we need this effect here could we do this by using a Route Guard?

Reply:
would be a big change but I think this whole signup flow should be done using machine instead of route/effect/navigate mixed with machines in some points.